pyccel/codegen/wrapper/c_to_python_wrapper.py

```diff
@@ -2562,6 +2565,87 @@ def _extract_HomogeneousSetType_FunctionDefArgument(self, orig_var, collect_arg,
 
         return {'body': body, 'args': arg_vars, 'clean_up': clean_up}
 
+    def _extract_StringType_FunctionDefArgument(self, orig_var, collect_arg, bound_argument,
+            is_bind_c_argument, *, arg_var = None):
+        """
+        Extract the C-compatible string FunctionDefArgument from the PythonObject.
+
+        Extract the C-compatible string FunctionDefArgument from the PythonObject.
+        The C-compatible argument is extracted from collect_arg which holds a Python
+        object into arg_var.
+
+        The extraction is done by allocating an array and filling the elements with values
+        extracted from the indexed Python tuple in collect_arg.
+
+        Parameters
+        ----------
+        orig_var : Variable | IndexedElement
+            An object representing the variable or an element of the variable from the
+            FunctionDefArgument being wrapped.
+
+        collect_arg : Variable
+            A variable with type PythonObject* holding the Python argument from which the
+            C-compatible argument should be collected.
+
+        bound_argument : bool
+            True if the argument is the self argument of a class method. False otherwise.
+            This should always be False for this function.
+
+        is_bind_c_argument : bool
+            True if the argument was saved in a BindCFunctionDefArgument. False otherwise.
+
+        arg_var : Variable | IndexedElement, optional
+            A variable or an element of the variable representing the argument that
+            will be passed to the low-level function call.
+
+        Returns
+        -------
+        list[PyccelAstNode]
+            A list of expressions which extract the argument from collect_arg into arg_var.
+        """
+        assert not bound_argument
```

Comment:
I find this more readable and safer (because it does not rely on `__bool__`):
```suggestion
        assert bound_argument is False
```

Reply:
Fixed [570b49f](https://github.com/pyccel/pyccel/pull/2200/commits/570b49f74b6374cd36f935095c47e29d216051c3)